compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -2079,4 +2079,17 @@ object messages {
     }
     val explanation = ""
   }
+
+  case class DoubleDeclaration(decl: Symbol, previousSymbol: Symbol)(implicit ctx: Context) extends Message(DoubleDeclarationID) {
+    val kind = "Duplicate Symbol"
+    val msg = {
+      val details = decl.asTerm.signature.matchDegree(previousSymbol.asTerm.signature) match {
+        case Signature.NoMatch => "" // matchDegree also returns NoMatch if one of the terms is not a method
```

Comment:
This sounds plausible, but I'm not sure how it relates to `realMethod` and why that code is there — not within my "review budget" I fear. I really want to approve this PR but I'm not sure it's not introducing regressions. Finding another reviewer.

Reply:
OTOH, not sure `isRealMethod` worked anyway: for `abstract class A(val a: Int) { def a: String }` on master I get:
```
scala> abstract class A(val a: Int) { def a: String }
1 |abstract class A(val a: Int) { def a: String }
  |                                   ^
  |                           method a is already defined as value a: Int
  |                           (the definitions have matching type signatures)
```